### PR TITLE
SONARIAC-1212 S6865: Should not raise an issue with an accompanied ServiceAccount

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

**Comment by rudy-regazzoni-sonarsource** [comment]
I extracted from [AbstractResourceManagementCheck.java](https://github.com/SonarSource/sonar-iac/pull/1388/files#diff-4c1aaba66b14a3e0927e6e5bb6b81d5f5325d3a74e7cdb4b2adc4b6920d990ef) the logics that I identified to be related to `LimitRange` elements and move it to [AbstractLimitCheck.java](https://github.com/SonarSource/sonar-iac/pull/1388/files#diff-37730ca779a72d289fcb31831ffd0b6e737f4d6f1c9cc5cb7b0a59ed6b7fc6f4).
My goal was to keep this class very general and usable for any `ProjectResource` element.

File: iac-common/src/main/java/org/sonar/iac/common/api/checks/CheckContext.java [hunk 1]
```diff
@@ -19,10 +19,11 @@
  */
 package org.sonar.iac.common.api.checks;
 
-import java.util.List;
 import org.sonar.iac.common.api.tree.HasTextRange;
 import org.sonar.iac.common.api.tree.impl.TextRange;
 
+import java.util.List;
```
Comment (petertrr) [inline iac-common/src/main/java/org/sonar/iac/common/api/checks/CheckContext.java:25]: I wonder why imports are not rearranged like this on my machine. Do you have `optimize imports` with an `importorder` file enabled in the Eclipse formatter adapter?

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/model/ServiceAccount.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import org.sonar.iac.common.checks.Trilean;
 
 public record ServiceAccount(
+  String path,
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/model/ServiceAccount.java:27]: Maybe call it `filePath`, to clearly indicate that it is related to the filesystem and not to some Kubernetes concept?

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/AutomountServiceAccountTokenCheck.java [hunk 3]
```diff
@@ -47,18 +51,61 @@ void registerObjectCheck() {
...
+        return;
+      }
+
+      List<ServiceAccount> linkedServiceAccounts = retrieveLinkedServiceAccount(blockObject);
+      if (linkedServiceAccounts.isEmpty()) {
+        specAsAttributeObject.reportOnKey(message);
+      } else {
+        boolean hasAtLeastOneAccountCompliant = linkedServiceAccounts.stream().anyMatch(account -> account.automountServiceAccountToken().isFalse());
+        if (!hasAtLeastOneAccountCompliant) {
+          // report on first linked account - there should be only one
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/AutomountServiceAccountTokenCheck.java:76]: Maybe, a combination of `filter` + `findFirst` + `isPresent` could be better here, than `anyMatch` + `get(0)`?

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/AbstractResourceManagementCheck.java [hunk 4]
```diff
@@ -21,58 +21,39 @@
...
 import static org.sonar.iac.common.yaml.TreePredicates.isSetString;
 
-public abstract class AbstractResourceManagementCheck extends AbstractKubernetesObjectCheck {
+public abstract class AbstractResourceManagementCheck<T extends ProjectResource> extends AbstractKubernetesObjectCheck {
   protected static final String KIND_POD = "Pod";
-  protected static final List<String> KIND_WITH_TEMPLATE = List.of(
-    "DaemonSet", "Deployment", "Job", "ReplicaSet", "ReplicationController", "StatefulSet", "CronJob");
+  protected static final List<String> KIND_WITH_TEMPLATE = List.of("DaemonSet", "Deployment", "Job", "ReplicaSet", "ReplicationController", "StatefulSet", "CronJob");
   protected static final Set<String> LIMIT_RANGE_LIMIT_TYPES = Set.of("Pod", "Container");
+  private String namespace;
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/AbstractResourceManagementCheck.java:41]: I like the idea of encapsulating the namespace in the superclass!

File: iac-extensions/kubernetes/src/test/resources/checks/MemoryLimitCheck/memory_limit_pod_with_global_limit.yaml [hunk 5]
```diff
@@ -3,7 +3,7 @@ apiVersion: v1
 kind: Pod
 metadata:
   name: example
-  namespace: ${namespace}
+  namespace: with-global-limit
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/test/resources/checks/MemoryLimitCheck/memory_limit_pod_with_global_limit.yaml:6]: In the `MemoryLimitCheckTest.testPodKindWithNamespace` this was being replaced by different namespaces. What's the reason for this change?

**Review by petertrr — APPROVED** [review]
LGTM, please address my suggestions before merging

File: iac-common/src/main/java/org/sonar/iac/common/api/checks/CheckContext.java (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-common/src/main/java/org/sonar/iac/common/api/checks/CheckContext.java:25]: I will check with other squad member so that I can align 👍

File: iac-extensions/kubernetes/src/test/resources/checks/MemoryLimitCheck/memory_limit_pod_with_global_limit.yaml (hunk 5 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/kubernetes/src/test/resources/checks/MemoryLimitCheck/memory_limit_pod_with_global_limit.yaml:6]: I believe this is a mistake, I will remove this.
